app/models/ability.rb

```diff
@@ -0,0 +1,20 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new
+    if user.role == 'admin'
+      can :manage, :all
+    else
+      can :destroy, Post do |post|
```

Comment:
- [ ] Nice work done on your ability class, when I tested it, I noticed that I can delete posts and comments has a user, you can modify it like below to make it work.

```
class Ability
  include CanCan::Ability

  def initialize(user)
    user ||= User.new
    if user.role == 'admin'
      can :manage, :all
    else
      can :create, Post
      can :create, Comment
      can :read, :all
    end
  end
end ```

Reply:
Hello, Please note that the instruction states that the user who is the author of a post or comment can delete it. An Admin (User with an Admin role) can also delete any post or comment.

The part you asked me to remove will not enable the author to delete posts or comments, which is against the requirement.

On this note, I would resubmit the project because your suggestion only gives a right to the admin and disallows the author.